### PR TITLE
Combine two "for" loops in "mergeAPIResourceConfigs" function

### PR DESCRIPTION
**Pull request by ping035627**
The PR combine two "for" loops in "mergeAPIResourceConfigs" function, it needn't loop twice, and I think it will be more effective.

### CODE REVIEW

**Comment by ping035627** [comment]
@apelisse  this pr make "do-not-merge"  by  robot to.

**Comment by ping035627** [comment]
@lavalamp
can test this pr? thanks.

**Comment by lavalamp** [comment]
ok to test

**Comment by ping035627** [comment]
@deads2k
 can you review this pr? need lgtm, thanks.

File: pkg/genericapiserver/default_storage_factory_builder.go [hunk 1]
```diff
@@ -101,51 +101,34 @@ func mergeAPIResourceConfigs(defaultAPIResourceConfig *ResourceConfig, resourceC
...
-	// Iterate through all group/version/resource overrides specified in runtimeConfig.
-	for key := range overrides {
-		tokens := strings.Split(key, "/")
-		if len(tokens) != 3 {
-			continue
-		}
-		groupVersionString := tokens[0] + "/" + tokens[1]
-		// HACK: Hack for "v1" legacy group version.
-		// Remove when we stop supporting the legacy group version.
-		if groupVersionString == "api/v1" {
```
Comment (deads2k) [inline pkg/genericapiserver/default_storage_factory_builder.go]: looks like you lost this.

**Comment by deads2k** [comment]
This doesn't seem to be functionally equivalent and given the difference in logic, I think prefer two loop construction.

File: pkg/genericapiserver/default_storage_factory_builder.go (hunk 1 above)
Comment (ping035627) [inline pkg/genericapiserver/default_storage_factory_builder.go]: Perhaps  you have losed the next codes, it  has  been dealed  like this,   here needn't  two loops, I think. 
	length := len(tokens)
		if length != 2 && length != 3 {
			continue
		}
		groupVersionString := tokens[0] + "/" + tokens[1]
		// HACK: Hack for "v1" legacy group version.
		// Remove when we stop supporting the legacy group version.
		if groupVersionString == "api/v1" {
			groupVersionString = "v1"
		}

**Comment by ping035627** [comment]
@deads2k
it's functionally equivalent,  please  check  the whole  "for" loop,  thanks.

File: pkg/genericapiserver/default_storage_factory_builder.go (hunk 1 above)
Comment (deads2k) [inline pkg/genericapiserver/default_storage_factory_builder.go]: Ah, got stripped in the diff.

**Comment by deads2k** [comment]
> @deads2k
> it's functionally equivalent, please check the whole "for" loop, thanks.

Sorry, I still think that his two loop construction was cleaner.  "handle things like X, then handle things like Y".

**Pull request closed without merging**